Test/Invoke-WhiskeyPester3Task.Tests.ps1

```diff
@@ -66,246 +197,149 @@ function Assert-PesterRan
+    GivenPesterPath -pesterPath 'PassingTests' 
+    WhenPesterTaskIsInvoked -ErrorAction SilentlyContinue
+    ThenPesterShouldHaveRun -PassingCount 0 -FailureCount 0
+    ThenTestShouldFail -failureMessage 'does not exist'
 }
 
 Describe 'Invoke-WhiskeyPester3Task.when major version of tool is not 3.*' {
     GivenTestContext
-    $version = '4.0.1'
-    $failureMessage = 'the major version number must always be ''3'''
-    Invoke-PesterTest -Path $pesterPassingPath -Version $version -ShouldFailWithMessage $failureMessage -PassingCount 0 -FailureCount 0 -ErrorAction SilentlyContinue
+    GivenVersion '4.0.1'
+    GivenPesterPath -pesterPath 'PassingTests' 
+    WhenPesterTaskIsInvoked -ErrorAction SilentlyContinue
+    ThenPesterShouldHaveRun -PassingCount 0 -FailureCount 0
+    ThenTestShouldFail -failureMessage 'the major version number must always be ''3'''
+
 }
 
 Describe 'Invoke-WhiskeyPester3Task.when Find-Module fails' {
     GivenTestContext
-    Mock -CommandName 'Find-Module' -ModuleName 'Whiskey' -MockWith { return $Null }
-    Mock -CommandName 'Where-Object' -ModuleName 'Whiskey' -MockWith { return $Null }
-    $failureMessage = 'Unable to find a version of Pester 3 to install.'
-    Invoke-PesterTest -Path $pesterPassingPath -FailureCount 0 -PassingCount 0 -WithMissingVersion -ShouldFailWithMessage $failureMessage -ErrorAction SilentlyContinue
-    Assert-MockCalled -CommandName 'Find-Module' -Times 1 -ModuleName 'Whiskey'
-    Assert-MockCalled -CommandName 'Where-Object' -Times 1 -ModuleName 'Whiskey'
+    GivenFindModuleFails
+    GivenPesterPath -pesterPath 'PassingTests'
+    WhenPesterTaskIsInvoked -ErrorAction SilentlyContinue
+    ThenPesterShouldHaveRun -FailureCount 0 -PassingCount 0
+    ThenFindModuleShouldHaveBeenCalled
+    ThenTestShouldFail -failureMessage 'Unable to find a version of Pester 3 to install.'
 }
 
 Describe 'Invoke-WhiskeyPester3Task.when a task path is absolute' {
     GivenTestContext
-    $Global:Error.Clear()
-    $path = 'C:\FubarSnafu'
-    $failureMessage = 'absolute'
-    Invoke-PesterTest -Path $path -ShouldFailWithMessage $failureMessage -PassingCount 0 -FailureCount 0 -ErrorAction SilentlyContinue
+    GivenPesterPath -pesterPath 'C:\FubarSnafu'
+    GivenVersion '3.4.3'
+    WhenPesterTaskIsInvoked -ErrorAction SilentlyContinue
+    ThenPesterShouldHaveRun -PassingCount 0 -FailureCount 0
+    ThenTestShouldFail -failureMessage 'absolute'
 }
 
 Describe 'Invoke-WhiskeyPester3Task.when running passing Pester tests with Clean Switch' {
     GivenTestContext
-    Invoke-PesterTest -Path $pesterPassingPath -FailureCount 0 -PassingCount 0 -withClean
+    GivenPesterPath -pesterPath 'PassingTests'
+    GivenVersion '3.4.3'
+    GivenWithCleanFlag
+    WhenPesterTaskIsInvoked
+    ThenPesterShouldHaveRun -FailureCount 0 -PassingCount 0
+    ThenTestShouldPass -withClean
+}
+
+Describe 'Invoke-WhiskeyPester3Task.when running passing Pester tests with initialization switch' {

```

Comment:
Add assertions that pester was installed and that no tests were run (i.e. no rest result files). Ditto for the Pester3 tests.

Reply:
Done for both

Reply:
I don't see any assertions that Pester was installed. Ditto for the Pester4 tests. Also, the tests shouldn't pass (`ThenTestShouldPass`) because they shouldn't be running. What does that assertion test?

Reply:
That is what the `ThenTestShouldPass` does, it checks that that Pester is installed,if you pass in `-withclean` it checks that  it is uninstalled. Maybe this should be refactored into two tests one `ThenPesterIsInstalled` and `ThenPesterIsUninstalled` to be more clear.